dts/riscv/rapidsilicon/rapidsi_virgo.dtsi

```diff
@@ -40,19 +40,26 @@
 			 0xE2001000 0x1000>;
 	 };
 	 
-	 sram: memory@A0300000 {
-		 compatible = "rapidsi,dlm";
-		 device_type = "memory";
-		 reg = <0xA0300000 DT_SIZE_K(32)>;
-		 status = "okay";
-	 };
-	 
-	 ilm: ilm@A0200060 {
-		 compatible = "rapidsi,ilm";
-		 device_type = "memory";
-		 reg = <0xA0200060 (DT_SIZE_K(64)-0x00000060)>;
-		 status = "okay";
-	 };
+	 memory: memory {
+		compatible = "rapidsi,ocm";
+		#address-cells = <1>;
+		#size-cells = <1>;
+		ranges;
+
+		dlm: memory@A0300000 {
+			compatible = "rapidsi,dlm";
+			device_type = "memory";
+			reg = <0xA0300000 DT_SIZE_K(32)>;
+			status = "okay";
+		};
+		
+		ilm: memory@A0200060 {
```

Comment:
We can definitely change it. Our FSBL binary is run from 0xA020005C because the first 5C bytes are taken by UBI header. The Zephyr binary address is kept 0xA0200060 due to alignment issues and also considering UBI header space. 

If we know how Zephyr will be put in ILM, we can then think of its execution location accordingly.

Reply:
> We can definitely change it. Our FSBL binary is run from 0xA020005C because the first 5C bytes are taken by UBI header. The Zephyr binary address is kept 0xA0200060 due to alignment issues and also considering UBI header space.
> 
> If we know how Zephyr will be put in ILM, we can then think of its execution location accordingly.

Ok, I understand you are suggesting that the binary will consist of the UBI header and the actual Zephyr binary too and we will be loading the entire binary into the ILM. But what about the BOP header of the zephyr binary. 

Zephyr should be loaded the same way as FSBL. So the ways of loading are 
1. Via openOCD: Do we currently load the entire binary (UBI binary with BOP or only the zephyr binary?) and then jump to the FSBL (in this case zephyr)?
2. Via UART: where the binary is loaded through UART and then jump the the binary. Same question as above
3. From Flash: In this case it is going to the the entire UBI binary

Reply:
At the moment, we are loading Zephyr using xmodem command on CLI over USART. 

For loading Zephyr via Flash, we will have to use the same method which we use for FPGA bitstreams where FSBL loads everything on the flash. Then we reboot into SPI boot mode and bootrom then loads Zephyr, after security stuff, into the ILM.